pkg/pquerier/select.go

```diff
@@ -374,6 +376,10 @@ func (queryCtx *selectQueryContext) createColumnSpecs() ([]columnMeta, map[strin
 	}
 
 	for i, col := range requestedColumns {
+		if col.Metric != "" && !queryCtx.existingMetrics[col.Metric] {
```

Comment:
Not critical, but `existingMetrics` would be clearer as an argument rather than receiver state.